src/pages/networks/forms/NetworkLocalPeeringForm.tsx

```diff
@@ -0,0 +1,220 @@
+import type { FC, ReactNode } from "react";
+import { Form, Input } from "@canonical/react-components";
+import type { FormikProps } from "formik/dist/types";
+import AutoExpandingTextArea from "components/AutoExpandingTextArea";
+import type { LxdNetwork } from "types/network";
+import { useNetworkEntitlements } from "util/entitlements/networks";
+import NetworkSelector from "pages/projects/forms/NetworkSelector";
+import { useNetworks } from "context/useNetworks";
+import { typesWithLocalPeerings } from "util/networks";
+import { useProjects } from "context/useProjects";
+import ProjectSelector from "./ProjectSelector";
+
+export interface LocalPeeringFormValues {
+  name: string;
+  targetProject: string;
+  targetNetwork: string;
+  description?: string;
+  customTargetProject?: string;
+  customTargetNetwork?: string;
+  createMutualPeering?: boolean;
+}
+
+interface Props {
+  formik: FormikProps<LocalPeeringFormValues>;
+  network: LxdNetwork;
+  isEditing?: boolean;
+}
+
+const NetworkLocalPeeringForm: FC<Props> = ({ formik, network, isEditing }) => {
+  const { canEditNetwork } = useNetworkEntitlements();
+  const { data: projects = [] } = useProjects();
+  const { data: networks = [] } = useNetworks(
+    formik.values.targetProject || "default",
+  );
+  const projectOtherLabel = "Manually enter project";
+  const networkOtherLabel = "Manually enter network";
+  const projectList = [
+    ...projects,
+    { name: projectOtherLabel, config: {}, description: "" },
+  ];
+
+  const managedNetworks = networks
+    .filter((item) => typesWithLocalPeerings.includes(item.type))
+    .filter((item) => item.name !== network.name);
+
+  const networkList = [
+    ...managedNetworks,
+    { name: networkOtherLabel, type: "ovn", config: {} } as LxdNetwork,
+  ];
+
+  const getFormProps = (
+    id:
+      | "name"
+      | "description"
+      | "targetProject"
+      | "targetNetwork"
+      | "customTargetProject"
+      | "customTargetNetwork",
+  ) => {
+    return {
+      id: id,
+      name: id,
+      onBlur: formik.handleBlur,
+      onChange: formik.handleChange,
+      value: formik.values[id] ?? "",
+      error: formik.touched[id] ? (formik.errors[id] as ReactNode) : null,
+      placeholder: `Enter ${id.replaceAll("-", " ")}`,
+    };
+  };
+
+  const editRestriction = canEditNetwork(network)
+    ? ""
+    : "You do not have permission to edit this network";
+
+  return (
+    <Form
+      onSubmit={formik.handleSubmit}
+      className={"local-peering-create-form"}
+    >
+      {/* hidden submit to enable enter key in inputs */}
+      <Input type="submit" hidden value="Hidden input" />
+      <Input
+        {...getFormProps("name")}
+        type="text"
+        label="Name"
+        required
+        autoFocus
```

Comment:
```suggestion
        autoFocus={!isEditing}
```
On edit the description field should receive the focus.